src/front/domain_socket.rs

```diff
@@ -248,3 +248,144 @@ impl DomainSocketListenerBuilder {
         })?)
     }
 }
+
+impl GetMetadata for UnixStream {
+    fn metadata(&self) -> Option<ConnectionMetadata> {
+        let ucred = peer_credentials::peer_cred(self)
+            .map_err(|err| {
+                format_error!(
+                    "Failed to grab peer credentials metadata from UnixStream",
+                    err
+                )
+            })
+            .ok()?;
+
+        Some(ConnectionMetadata::UnixPeerCredentials {
+            uid: ucred.uid,
+            gid: ucred.gid,
+        })
+    }
+}
+
+// == IMPORTANT NOTE ==
+//
+// The code below has been cherry-picked from the following PR:
+//
+//     https://github.com/rust-lang/rust/pull/75148
```

Comment:
Could you also please add the stabilizing issue https://github.com/rust-lang/rust/issues/42839?

Reply:
Done.